packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -203,20 +139,132 @@ class DirectStripeRoutes {
   }
 
   /**
-   * Verify that the invoice was paid successfully.
+   * Create Subscription for New Customer
+   *
+   * @param {string} uid
+   * @param {string} email
+   * @param {string} displayName
+   * @param {string} paymentToken
+   * @param {AbbrevPlan} selectedPlan
+   *
+   * @returns {Promise<Subscription>}
+   */
+  async createSubscriptionNewCustomer(
+    uid,
+    email,
+    displayName,
+    paymentToken,
+    selectedPlan
+  ) {
+    let customer;
+
+    try {
+      customer = await this.stripeHelper.createCustomer(
+        uid,
+        email,
+        displayName,
+        paymentToken
+      );
+    } catch (err) {
+      if (err.type === 'StripeCardError') {
+        throw error.rejectedSubscriptionPaymentToken(err.message, err);
+      }
+      throw err;
+    }
+
+    const subscription = await this.stripeHelper.createSubscription(
+      customer,
+      selectedPlan
+    );
+
+    return subscription;
+  }
+
+  /**
+   * Create Subscription for Existing Customer
    *
-   * Note that the invoice *must have the `payment_intent` expanded*
-   * or this function will fail.
+   * @param {Customer} customer
+   * @param {string} paymentToken
+   * @param {AbbrevPlan} selectedPlan
    *
+   * @returns {Promise<Subscription>}
+   */
+  async createSubscriptionExistingCustomer(
+    customer,
+    paymentToken,
+    selectedPlan
+  ) {
+    if (paymentToken) {
+      // Always update the source if we are given a paymentToken
+      // Note that if the customer already exists and we were not
+      // passed a paymentToken value, we will not update it and use
+      // the default source.
+      try {
+        await this.stripeHelper.updateCustomerPaymentMethod(
+          customer.id,
+          paymentToken
+        );
+      } catch (err) {
+        if (err.type === 'StripeCardError') {
+          throw error.rejectedSubscriptionPaymentToken(err.message, err);
+        }
+        throw err;
+      }
+    }
+
+    // Check if the customer already has subscribed to this plan.
+    let subscription = this.findCustomerSubscriptionByPlanId(
+      customer,
+      selectedPlan.plan_id
+    );
+
+    // If we have a prior subscription, we have 3 options:
+    //   1) Open subscription that needs a payment method, try to pay it
+    //   2) Paid subscription, stop and return as they already have the sub
+    //   3) Old subscription will have no open invoices, ignore it
+    if (subscription && subscription.latest_invoice) {
+      const invoice = /** @type {Invoice} */ (subscription.latest_invoice);
+      if (invoice.status === 'open') {
+        await this.handleOpenInvoice(invoice);
+        return subscription;
+      } else if (invoice.status === 'paid') {
+        throw error.subscriptionAlreadyExists();
+      }
+    }
+
+    subscription = await this.stripeHelper.createSubscription(
```

Comment:
Same thing here for catching StripeCardErrors. Then again, @bbangert had mentioned wrapping the entire route in an error handler, so that might be a smarter approach.